doc/sphinx/proof-engine/ltac.rst

```diff
@@ -192,14 +172,16 @@ evaluation yields either a term, an integer or a tactic. Intermediate
 results can be terms or integers but the final result must be a tactic
 which is then applied to the focused goals.
 
-There is a special case for ``match goal`` expressions of which the clauses
-evaluate to tactics. Such expressions can only be used as end result of
-a tactic expression (never as argument of a non-recursive local
+There is a special case for :tacn:`match goal` expressions of which the clauses
+evaluate to tactics. Such expressions can only be used as the end result of
+a tactic expression (never as an argument of a non-recursive local
 definition or of an application).
```

Comment:
I think this might be documenting the <= 8.4 behavior of tactics?

Reply:
What do you suggest here?  I have no knowledge about 8.4.

Reply:
Perhaps just remove it?  Consider the following change in behavior:
```coq
Goal True.
  let f tac := tac; tac in
  f ltac:(match goal with
          | [ |- True ] => idtac "True"
          | [ |- False ] => idtac "False"
          end).
  (* 8.4: Error: Immediate match producing tactics not allowed in local definitions. *)
  (* 8.5 and later: True *)
```
This error message is what this paragraph is talking about, and the error message no longer exists in >= 8.5.

I think also that the paragraph above this paragraph should be rewritten in any case.

Reply:
@JasonGross I'll drop the paragraph, however I don't know what to change in the preceeding paragraph.

Reply:
See https://github.com/coq/coq/pull/12103/files#r432990765 for a suggestion